panel_segmentation/panel_detection.py

```diff
@@ -10,529 +12,677 @@
+panel_classification_model_path = path.join(path.dirname(__file__),
+                                            'VGG16_classification_model.h5')
+mounting_classification_model_path = path.join(path.dirname(__file__),
+                                               'object_detection_model.pth')
 
-class PanelDetection():
+
+class PanelDetection:
     '''
-    A class for training a deep learning architecture, 
+    A class for training a deep learning architecture,
     detecting solar arrays from a satellite image, performing spectral
-    clustering, and predicting the Azimuth.
+    clustering, predicting azimuth, and classifying mounting type and
+    configuration.
     '''
-    def __init__(self, model_file_path = './VGG16Net_ConvTranpose_complete.h5', 
-                 classifier_file_path = './VGG16_classification_model.h5'):
-        
-        #This is the model used for detecting if there is a panel or not
-        self.classifier = load_model(classifier_file_path, 
-                                     custom_objects=None, 
+
+    def __init__(self, model_file_path='./VGG16Net_ConvTranpose_complete.h5',
+                 classifier_file_path='./VGG16_classification_model.h5',
+                 mounting_classifier_file_path='./object_detection_model.pth'):
+        # This is the model used for detecting if there is a panel or not
+        self.classifier = load_model(classifier_file_path,
+                                     custom_objects=None,
                                      compile=False)
-        
-        self.model = load_model(model_file_path, 
-                                custom_objects=None, 
+        self.model = load_model(model_file_path,
+                                custom_objects=None,
                                 compile=False)
-        
-        
-    def generateSatelliteImage(self,latitude, longitude, 
+        self.mounting_classifier = core.Model.load(
+            mounting_classifier_file_path, ["ground-fixed",
+                                            "carport-fixed",
+                                            "rooftop-fixed",
+                                            "ground-single_axis_tracker"])
+
+    def generateSatelliteImage(self, latitude, longitude,
                                file_name_save, google_maps_api_key):
         """
-        Generates satellite image via Google Maps, using the passed lat-long coordinates.
-        
+        Generates satellite image via Google Maps, using a set of lat-long
+        coordinates.
+
         Parameters
         -----------
-        latitude: Float. Latitude coordinate of the site.
-        longitude: Float. Longitude coordinate of the site.
-        file_name_save: String. File path that we want to save the image to. PNG file.
-        google_maps_api_key: String. Google Maps API Key for automatically 
-            pulling satellite images.
-                
+        latitude: (float)
```

Comment:
The parentheses around parameter types are visual clutter IMHO.  Note that sphinx adds parentheses automatically:

![image](https://user-images.githubusercontent.com/57452607/166061751-4d199573-cb36-407f-b0a8-ab5ef8317d1d.png)

Reply:
Removed these parentheses in commit https://github.com/NREL/Panel-Segmentation/pull/15/commits/c11b781b929615fa3d5646f42266f0285f8b0547